api/app/src/main/kotlin/packit/controllers/PacketController.kt

```diff
@@ -77,4 +81,29 @@ class PacketController(
             .headers(response.second)
             .body(response.first)
     }
+
+    @GetMapping("/{name}/{id}/zip")
+    @PreAuthorize("@authz.canReadPacket(#root, #id, #name)")
+    // TODO: Verify authorization for accessing specific files: whether user has access to all files, or artefacts only.
+    fun downloadZip(
+        @PathVariable id: String,
+        @PathVariable name: String,
+        @RequestParam(required = true) hashes: List<String>,
+        @RequestParam(required = true) filenames: List<String>,
+    ): ResponseEntity<StreamingResponseBody>
+    {
+        val filesWithNames = hashes.mapIndexed { index, hash ->
+            val fileResource = packetService.getFileByHash(hash, false, hash).first
```

Comment:
Does this read in the whole file? Probably want to deal with streaming across each file in turn in the zip service, some packet files are massive!.